pkg/http/http.go

```diff
@@ -0,0 +1,109 @@
+package http
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// Client defines an http client object
+type Client struct {
+	timeout  time.Duration
+	username string
+	password string
+	token    string
+}
+
+// ClientOptions defines the options to be set on the client
+type ClientOptions struct {
+	Timeout  time.Duration
+	Username string
+	Password string
+	Token    string
+}
+
+// Sender provides an interface to the piper http client for uid/pwd authenticated requests
+type Sender interface {
+	SendRequest(method, url string, body io.Reader, header http.Header, cookies []*http.Cookie) (*http.Response, error)
+	SetOptions(options ClientOptions)
+}
+
+// SendRequest sends an http request with a defined method
+func (c *Client) SendRequest(method, url string, body io.Reader, header http.Header, cookies []*http.Cookie) (*http.Response, error) {
+
+	logger := log.Entry().WithField("package", "github.wdf.sap.corp/ContinuousDelivery/piper-library/pkg/http")
```

Comment:
wrong url

Reply:
The package within the repository should be sufficient as the repository is a separate field: https://github.com/SAP/jenkins-library/blob/master/pkg/log/log.go#L14

Reply:
will change this ...